### PR TITLE
SONARJAVA-1836 Upgrade to SQ API 6.1

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: java-frontend/src/main/java/org/sonar/plugins/java/api/CheckRegistrar.java [hunk 1]
```diff
@@ -37,6 +38,7 @@
  * </pre>
  */
 @Beta
+@SonarLintSide
 @BatchSide
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/plugins/java/api/CheckRegistrar.java]: MINOR Remove this use of "BatchSide"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* MINOR 4 minor

Watch the comments in this conversation to review them.

File: java-checks/src/main/java/org/sonar/java/filters/PostAnalysisIssueFilter.java [hunk 2]
```diff
@@ -21,7 +21,6 @@
 
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.collect.ImmutableList;
-
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/filters/PostAnalysisIssueFilter.java]: useless change

**Pull request merged**